### PR TITLE
Attestation Fixes

### PR DESCRIPTION
**Pull request by nisdas**
Following #3105 
- [x] Remove AttestationDataFromSlot when saving attestation targets 
- [x] Allow attestation pools to filter slots by expected state slot.

### CODE REVIEW

File: beacon-chain/operations/service.go [hunk 1]
```diff
@@ -124,30 +125,35 @@ func (s *Service) IncomingProcessedBlockFeed() *event.Feed {
...
 	}
-	state, err := s.beaconDB.HeadState(ctx)
+	bState, err := s.beaconDB.HeadState(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("could not retrieve attestations from DB")
 	}
 
+	bState, err = state.ProcessSlots(ctx, bState, requestedSlot)
+	if err != nil {
+		return nil, fmt.Errorf("could not process slots %v", err)
```
Comment (terencechain) [inline beacon-chain/operations/service.go]:
```suggestion
		return nil, fmt.Errorf("could not process slots up to %d: %v", requestedSlot, err)
```

File: beacon-chain/rpc/attester_server.go [hunk 2]
```diff
@@ -126,7 +119,7 @@ func (as *AttesterServer) RequestAttestation(ctx context.Context, req *pb.Attest
 
 	headState, err = state.ProcessSlots(ctx, headState, req.Slot)
 	if err != nil {
-		return nil, fmt.Errorf("could not process slot: %v", err)
+		return nil, fmt.Errorf("could not process slots %v", err)
```
Comment (terencechain) [inline beacon-chain/rpc/attester_server.go]:
```suggestion
		return nil, fmt.Errorf("could not process slots up to %d: %v", req.Slot, err)
```

File: shared/params/config.go [hunk 3]
```diff
@@ -234,9 +234,9 @@ func MainnetConfig() *BeaconChainConfig {
 func DemoBeaconConfig() *BeaconChainConfig {
 	demoConfig := MinimalSpecConfig()
 	demoConfig.MinDepositAmount = 100
-	demoConfig.MaxEffectiveBalance = 3.2 * 1e9
-	demoConfig.EjectionBalance = 1.6 * 1e9
-	demoConfig.EffectiveBalanceIncrement = 0.1 * 1e9
+	demoConfig.MaxEffectiveBalance = 3.2 * 1e6
```
Comment (terencechain) [inline shared/params/config.go]: ???

Comment (nisdas) [reply shared/params/config.go]: oh , sorry . It was a config change for local testing. Reverting this

File: beacon-chain/rpc/proposer_server.go [hunk 4]
```diff
@@ -276,6 +277,7 @@ func (ps *ProposerServer) deposits(ctx context.Context, currentVote *ethpb.Eth1D
 	// root will fail to verify. This can happen in a scenario where we perhaps have a deposit from height 101,
 	// so we want to avoid any possible mismatches in these lengths.
 	upToEth1DataDeposits := ps.beaconDB.AllDeposits(ctx, latestEth1DataHeight)
+	log.Errorf("uptoEth1deps %d and alldeps %d at blockHeight %d", len(upToEth1DataDeposits), len(allDeps), latestEth1DataHeight.Uint64())
```
Comment (prestonvanloon) [inline beacon-chain/rpc/proposer_server.go]: Remove? is this a debug?

Comment (nisdas) [reply beacon-chain/rpc/proposer_server.go]: ah was for debugging, removing this

**Pull request merged**